style: apply /style-guide pass to sandboxes (#2668)
wandb/docs: sandboxes/run-commands.mdx

---
title: Run commands in a sandbox
description: Learn how to run commands in a sandbox environment and read output and exit codes.

> keywords: ["Sandbox.exec", "Process.result", "SandboxExecutionError", "stdout stderr"]

Reviewer: ```suggestion
keywords: ["Sandbox.run", "Sandbox.exec", "Process.result", "SandboxExecutionError", "stdout", "stderr"]
```

Author: Applied in 1159ffaa8.

<br></br>

> This page shows you how to [run commands in a sandbox](#run-commands-in-a-sandbox), [read their output and exit code](#read-output-and-exit-codes), and [check for errors](#check-for-errors). Use these features to run scripts, training jobs, and other processes in a sandbox and programmatically check the results.

Reviewer: ```suggestion
[Run commands in a sandbox](#run-commands-in-a-sandbox), [read command output and exit codes](#read-output-and-exit-codes), and [check for errors](#check-for-errors). Use these features to run scripts, training jobs, and other processes in a W&B Serverless Sandbox and programmatically check the results.
```

Removes "This page shows you how to..."

Author: Applied in 1159ffaa8. I also repointed the check-for-errors link to the real heading anchor (#check-for-sandbox-execution-errors) so it is not broken.

## Read output and exit codes

> After a command runs, you often need to read its output and check whether it succeeded. `Sandbox.exec()` returns a [`Process`](https://docs.coreweave.com/products/coreweave-sandbox/client/ref/types/process) object. Call `Process.result()` to wait for the command to finish and get its result. `Process.result()` includes the standard output (`stdout`), standard error (`stderr`), and exit code (`returncode`).

Reviewer: ```suggestion
After you run a command, you can read its output and check whether it succeeded. `Sandbox.exec()` returns a [`Process`](https://docs.coreweave.com/products/coreweave-sandbox/client/ref/types/process) object. To wait for the command to finish, call `Process.result()`.

The result includes standard output (`stdout`), standard error (`stderr`), and the exit code (`returncode`).
```

Author: Applied in 1159ffaa8.

## Check for sandbox execution errors

> The `check` parameter raises an exception when a command fails, so you don't need to inspect exit codes manually. Specify `check=True` (`Sandbox.exec(check=True)`) to raise a [`SandboxExecutionError`](https://docs.coreweave.com/products/coreweave-sandbox/client/ref/exceptions/exceptions#sandboxexecutionerror) if the command exits with a non-zero exit code. This example runs a script `might_fail.py` and checks the result.

Reviewer: ```suggestion
Use `check=True` to raise an exception when a command fails instead of inspecting exit codes manually. If the command exits with a non-zero exit code, `Sandbox.exec(check=True)` raises a [SandboxExecutionError](https://docs.coreweave.com/products/coreweave-sandbox/client/ref/exceptions/exceptions#sandboxexecutionerror).
```

Author: Applied in 1159ffaa8.
